quixstreams/sources/community/file/azure.py

```diff
@@ -0,0 +1,157 @@
+        shutdown_timeout: float = 30,
+        on_client_connect_success: Optional[ClientConnectSuccessCallback] = None,
+        on_client_connect_failure: Optional[ClientConnectFailureCallback] = None,
+    ):
+        """
+        :param connection_string: Azure client authentication string.
+        :param container: Azure container name.
+        :param filepath: folder to recursively iterate from (a file will be used directly).
+        :param key_setter: sets the kafka message key for a record in the file.
+        :param value_setter: sets the kafka message value for a record in the file.
+        :param timestamp_setter: sets the kafka message timestamp for a record in the file.
+        :param file_format: what format the files are stored as (ex: "json").
+        :param compression: what compression was used on the files, if any (ex. "gzip").
+        :param has_partition_folders: whether files are nested within partition folders.
+            If True, FileSource will match the output topic partition count with it.
+            Set this flag to True if Quix Streams FileSink was used to dump data.
+            Note: messages will only align with these partitions if original key is used.
+            Example structure - a 2 partition topic (0, 1):
+            [/topic/0/file_0.ext, /topic/0/file_1.ext, /topic/1/file_0.ext]
+        :param replay_speed: Produce messages with this speed multiplier, which
+            roughly reflects the time "delay" between the original message producing.
+            Use any float >= 0, where 0 is no delay, and 1 is the original speed.
+            NOTE: Time delay will only be accurate per partition, NOT overall.
+        :param name: The name of the Source application (Default: last folder name).
+        :param shutdown_timeout: Time in seconds the application waits for the source
+            to gracefully shut down.
+        :param on_client_connect_success: An optional callback made after successful
+            client authentication, primarily for additional logging.
+        :param on_client_connect_failure: An optional callback made after failed
+            client authentication (which should raise an Exception).
+            Callback should accept the raised Exception as an argument.
+            Callback must resolve (or propagate/re-raise) the Exception.
+
+        """
+        super().__init__(
+            filepath=filepath,
+            key_setter=key_setter,
+            value_setter=value_setter,
+            timestamp_setter=timestamp_setter,
+            file_format=file_format,
+            compression=compression,
+            has_partition_folders=has_partition_folders,
+            replay_speed=replay_speed,
+            name=name or f"azurefilestore_{container}_{Path(filepath).name}",
+            shutdown_timeout=shutdown_timeout,
+            on_client_connect_success=on_client_connect_success,
+            on_client_connect_failure=on_client_connect_failure,
```

Comment:
WDYT if we focus only on what is needed to make this subclass distinct? This should do, but I have no way to test it, so you'd have to do it. I understand if you prefer not to 😅

```suggestion
        name: Optional[str] = None,
        **kwargs
    ):
        """
        :param connection_string: Azure client authentication string.
        :param container: Azure container name.
        :param filepath: folder to recursively iterate from (a file will be used directly).
        :param name: The name of the Source application (Default: last folder name).
        :param kwargs: Additional keyword arguments passed to FileSource.
        """
        super().__init__(
            filepath=filepath,
            name=name or f"azurefilestore_{container}_{Path(filepath).name}",
            **kwargs
```

Reply:
In general I like the idea because I've always hated all the passthrough we've had to do, but we otherwise lose type hinting and auto-complete for the **kwargs, so I'm keeping this consistent with our usual approach as the unfortunate necessary evil haha.

That said, what I sent you in Slack outlines another way we could potentially achieve both, but I'm not sure if it's too much of a change. Take a look and lemme know what you think.

Reply:
You mean we lose type hint and auto-complete in the platform, right?

In this case, I think it's best to leave it as is. I'm not sure I like the `Unpack` workaround. Maintaining a separate class to describe `kwargs` seems even stranger 🤔